### PR TITLE
fix(sockets) always uncork when closing

### PR DESCRIPTION
**Pull request by cirospaciari**
### What does this PR do?
Make sure that sockets always uncorked when closed

- [ ] Documentation or TypeScript types (it's okay to leave the rest blank in this case)
- [x] Code changes

### How did you verify your code works?
Existing tests

### CODE REVIEW

**Comment by robobun** [comment]
@cirospaciari, your commit 2396c319a2d2fffd9f78ae118628fc31005579e3 is building: [`#1997`](https://buildkite.com/bun/bun/builds/1997)

File: packages/bun-uws/src/AsyncSocket.h [hunk 1]
```diff
@@ -166,33 +166,34 @@ struct AsyncSocket {
...
+                ourCorkOffset = loopData->getCorkOffset();
+                loopData->setCorkOffset(0);
             }
 
             /* Fallback is to use the backpressure as buffer */
             backPressure.resize(ourCorkOffset + existingBackpressure + size);
 
             /* And copy corkbuffer in front */
-            memcpy((char *) backPressure.data() + existingBackpressure, loopData->corkBuffer, ourCorkOffset);
+            memcpy((char *) backPressure.data() + existingBackpressure, loopData->getCorkBuffer(), ourCorkOffset);
```
Comment (Jarred-Sumner) [inline packages/bun-uws/src/AsyncSocket.h]: `ourCorkOffset` will never be zero right?

Comment (cirospaciari) [reply packages/bun-uws/src/AsyncSocket.h]: I think it can be in edge cases, still worth checking

File: packages/bun-uws/src/LoopData.h [hunk 2]
```diff
@@ -44,7 +44,11 @@ struct alignas(16) LoopData {
 
     /* Map from void ptr to handler */
     std::map<void *, MoveOnlyFunction<void(Loop *)>> postHandlers, preHandlers;
-
+    /* Cork data */
+    char *corkBuffer = new char[CORK_BUFFER_SIZE];
```
Comment (Jarred-Sumner) [inline packages/bun-uws/src/LoopData.h:48]: We would need to free this

If we instead make this statically allocated, we wouldn't need to worry about freeing it:
```suggestion
    char corkBuffer[CORK_BUFFER_SIZE];
```

Comment (Jarred-Sumner) [reply packages/bun-uws/src/LoopData.h:48]: oh this already existed

**Pull request merged**